tasks/main.yml

```diff
@@ -1,26 +1,19 @@
 ---
-- name: Configure the Docker APT key
-  apt_key: keyserver="{{ docker_keyserver }}"
-           id="58118E89F3A912897C070ADBF76221572C52609D"
-           state=present
+- name: Download Docker APT key
+  apt_key: url={{ docker_repository_url }}/gpg id=0EBFCD88 state=present
 
 - name: Configure the Docker APT repository
-  apt_repository: repo="deb https://apt.dockerproject.org/repo {{ ansible_distribution|lower }}-{{ ansible_distribution_release }} main"
+  apt_repository: repo="deb {{ docker_repository_url }} {{ ansible_distribution_release }} {{ item.channel }}"
                   state=present
-
-- name: Install AUFS dependencies
-  apt: pkg="{{ item }}" state=present
   with_items:
-    - "linux-image-extra-{{ ansible_kernel }}"
-    - "linux-image-extra-virtual"
-    - "aufs-tools"
-  when: docker_options | search("--storage-driver=aufs")
+    - { channel: stable, install: true }
+    - { channel: edge, install: "{{ docker_install_edge | bool }}" }
```

Comment:
I didn't realize the complexity involved in supporting the `edge` distribution channel. I think it'll probably be better for us to just stop support for it entirely.